### PR TITLE
display the time left as "Connecting" when the sync time is undefined

### LINKED ISSUE #855: Handle 0% sync or network down in overview for progress estimate
If the network is down or 0% of sync completed the dialog box shows a blank estimate with 'left' string. Replace the 'left' string with 'estimating' when the estimated time is undefined.

godcr-estimate

### PR DESCRIPTION
**Pull request by AdimekweEbuka**
Resolves #855 
godcr (testnet) 2022-03-31 14-59-36

### CODE REVIEW

File: ui/page/overview/sync_details.go [hunk 1]
```diff
@@ -135,7 +136,11 @@ func (pg *AppOverviewPage) progressStatusRow(gtx C, inset layout.Inset) D {
 	}
 
 	percentageLabel := pg.Theme.Body1(fmt.Sprintf("%v%%", progress))
-	timeLeftLabel := pg.Theme.Body1(fmt.Sprintf("%v left", timeLeft))
+	if progress == 0 {
+		timeLeftLabel = pg.Theme.Body1("Estimating")
```
Comment (dreacot) [inline ui/page/overview/sync_details.go]:
```suggestion
		timeLeftLabel = pg.Theme.Body1("estimating")
```

**Review by dreacot — DISMISSED** [review]

File: ui/page/overview/sync_details.go (hunk 1 above)
Comment (dreacot) [reply ui/page/overview/sync_details.go]: according to the referenced issue

Comment (monsa00) [reply ui/page/overview/sync_details.go]: Hello, after meeting with @raedah we've decided to replace `Estimating` with `Connecting` please.

File: ui/page/overview/sync_details.go [hunk 2]
```diff
@@ -126,6 +126,7 @@ func (pg *AppOverviewPage) progressBarRow(gtx C, inset layout.Inset) D {
 
 // progressStatusRow lays out the progress status when the wallet is syncing.
 func (pg *AppOverviewPage) progressStatusRow(gtx C, inset layout.Inset) D {
+	var timeLeftLabel decredmaterial.Label
```
Comment (Sirmorrison) [inline ui/page/overview/sync_details.go]: remove

File: ui/page/overview/sync_details.go [hunk 3]
```diff
@@ -135,7 +136,11 @@ func (pg *AppOverviewPage) progressStatusRow(gtx C, inset layout.Inset) D {
...
 	percentageLabel := pg.Theme.Body1(fmt.Sprintf("%v%%", progress))
-	timeLeftLabel := pg.Theme.Body1(fmt.Sprintf("%v left", timeLeft))
+	if progress == 0 {
+		timeLeftLabel = pg.Theme.Body1("connecting")
+	} else {
+		timeLeftLabel = pg.Theme.Body1(fmt.Sprintf("%v left", timeLeft))
+	}
 	return inset.Layout(gtx, func(gtx C) D {
 		return components.EndToEndRow(gtx, percentageLabel.Layout, timeLeftLabel.Layout)
 	})
```
Comment (Sirmorrison) [inline ui/page/overview/sync_details.go]:
```suggestion
        timeLeft := fmt.Sprintf("%v left", timeLeft)
	if progress == 0 {
		timeLeft = ("connecting")
	}
	
	return inset.Layout(gtx, func(gtx C) D {
		return components.EndToEndRow(gtx, percentageLabel.Layout, pg.Theme.Body1(timeLeft).Layout)
	})
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**